src/app/CommandSender.cpp

```diff
@@ -128,6 +128,23 @@ CHIP_ERROR CommandSender::SendCommandRequestInternal(const SessionHandle & sessi
     return SendInvokeRequest();
 }
 
+void CommandSender::CreateBackupForRequestRollback(RollbackData & aRollbackData)
+{
+    VerifyOrReturn(mBufferAllocated);
+    VerifyOrReturn(mState == State::Idle || mState == State::AddedCommand);
+    VerifyOrReturn(mInvokeRequestBuilder.GetInvokeRequests().GetError() == CHIP_NO_ERROR);
+    VerifyOrReturn(mInvokeRequestBuilder.GetError() == CHIP_NO_ERROR);
```

Comment:
since RollbackData checkpoint is passed in *this, I wonder if all these checks should be there? Then we seem to not even need a CreateBackupForRequestRollback as a separate entry.

Otherwise maybe we need to define why some are here vs others inside Checkpoint.